### ISSUE TITLE
Backspace at end of empty line doesn't delete whole line and go to end of above line

### ISSUE THREAD

**Issue by mccreeker** [Microsoft/vscode#42785]
- VSCode Version: 1.19.2
- OS Version: W10LTSB

Steps to Reproduce:

1. Create a new line of a block of multi-line code like an opening div tag or JavaScript or PHP function, then hit Enter.
2. If you have an indent, continue to step 3. If not, create an indent with Tab or 4 spaces (whatever your religion is).
3. Hit the Backspace key; notice it merely removes an indent each time instead of deleting the whole line and putting your cursor at the end of the previous line like awesome JetBrains software does.
4. Continue to use VS Code because nobody can afford to buy another JetBrains license for your new job.

This should be standard logic in all editors; there's zero functional or stylistic reason to want to just delete indents on an empty line with Backspace (unless there's too many indents there which is rare, or your editor auto-indented wrong), and even if for some Pan-like reason you need to use that regressive space, you can just use Ctrl+Home to delete the line then indent where you want (or Shift+arrow, or just arrow if you're a weirdo).

Does this issue occur when all extensions are disabled?: Yes

**Comment by rmunn** [comment]
I like this feature idea, but one other thing to think about is the interaction with the `editor.trimAutoWhitespace` setting. If `editor.trimAutoWhitespace` is false, then I might want to delete some whitespace after hitting Enter, while still keeping the new line. And in those situations, reaching for the Backspace key would feel like the most natural way to do it.

So if this is implemented, I'd like to see a setting to turn it off for the rare cases when I might actually need to backspace over some auto-inserted whitespace (e.g., if I've turned `editor.trimAutoWhitespace` off for some reason).

**Comment by mccreeker** [comment]
Auto-adding whitespace should happen only after hitting Enter on an indented line, and only be added up to the beginning of the previous line or one indent more if it's inside a nesting (or maybe two more such as inside an HTML tag or PHP array so that the inner contents aren't on the same line as the following block segment), and I could see how it would be useful to just delete the empty whitespace on the line (instead of deleting the line), but this means two Backspaces to delete the line.

It's easy to see how Eclipse, JetBrains, and NetBeans editors handle Enter indentation and Backspace on empty or indent-only lines, though the JetBrains way has been the most useful and avoided the most keystrokes in my usage of all code editors I've used.

**Comment by vasilev-alex** [comment]
That is something that should be implemented natively in VS Code. But for now I found this extension really useful https://marketplace.visualstudio.com/items?itemName=jasonlhy.hungry-delete

Saves up hours a day :)

**Comment by rmunn** [comment]
If #71506 is implemented, that would probably solve this one as well.

**Comment by ryankhart** [comment]
This extension seems like it would solve your issue. It worked for me.
https://marketplace.visualstudio.com/items?itemName=jasonlhy.hungry-delete

**Comment by waleedrana777** [comment]
+1

**Comment by edgarwideman** [comment]
Still need this added!

**Comment by ryankhart** [comment]
@edgarwideman Have you tried this extension yet? It worked almost as good as how the JetBrains IDEs do it.
https://marketplace.visualstudio.com/items?itemName=jasonlhy.hungry-delete

**Comment by edgarwideman** [comment]
I did find it after dropping previous comment. Thanks anyway. 👍

**Comment by starball5** [comment]
Related on Stack Overflow: [Move cursor up after deleting a line in VSCode](https://stackoverflow.com/q/52638937/11107541)

**Comment by ArturoDent** [comment]
I posted a possible solution on SO: https://stackoverflow.com/a/78290422/836330

![deleteEmptyLinesMoveCursorUp](https://github.com/microsoft/vscode/assets/31813701/218bf4aa-3423-4a51-a95b-fc12a69a7286)

It is just a keybinding, no extension neccessary.

**Comment by NatoBoram** [comment]
Would be nice to have this natively as [extensions can be buggy](https://github.com/Jasonlhy/VSCode-Hungry-Delete/issues/44)

**Comment by NinjaZMY** [comment]
This seems like a fictional problem, as text editors are supposed to delete one character at a time using backspace. 
Also the problem looks solved via a keybinding, which would enable this feature, or just tweaking JSON commands. 
You can already use the proposed solutions online <see above> or use any extension that enables similar features

**Comment by ryankhart** [comment]
> text editors are supposed to delete one character at a time using backspace

says someone who's never used a JetBrains IDE.
